### PR TITLE
BLD: use pyproject extras to match new ci-helpers

### PR DESCRIPTION
**Pull request by tangkong**
## Description
Updates build requirements to match updated CI-Helpers from https://github.com/pcdshub/pcds-ci-helpers/pull/127

## Motivation and Context
Improve builds and testing in CI.

## How Has This Been Tested?
Looking for green checks in CI.

## Where Has This Been Documented?
This PR

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran ``docs/pre-release-notes.sh`` and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by tangkong** [comment]
Conda 3.11 fails to build the env, attempting to install pytest 2.7 (??!?)

**Comment by ZLLentz** [comment]
> attempting to install pytest 2.7

Sometimes these compatibility solvers repeatedly try older and older versions of a library when none of the new ones are marked as compatible which can make analyzing the build output difficult. This problem is compounded by an observation that old versions of a package are more likely to incomplete or incorrect/out of date dependency information.

File: docs-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+docs-versions-menu
+doctr
```
Comment (klauer) [inline docs-requirements.txt:2]:
```suggestion
```

**Review by klauer — APPROVED** [review]

**Comment by ZLLentz** [comment]
CI failure is the recurring race condition

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**